### PR TITLE
global: encoding detection fixes

### PR DESCRIPTION
**Pull request by slint**

### CODE REVIEW

File: invenio_previewer/utils.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    :param fp: Open Python file pointer.
+    :param default: Fallback encoding to use.
+    :returns: The detected encoding.
+    """
+    try:
+        sample = fp.read(
+            current_app.config.get('PREVIEWER_CHARDET_BYTES', 1024))
+        return chardet.detect(sample).get('encoding', default)
+    except Exception:
+        return default
```
Comment (jirikuncar) [inline invenio_previewer/utils.py]: shall we add some logging here?

File: invenio_previewer/config.py [hunk 2]
```diff
@@ -27,6 +27,8 @@
 PREVIEWER_CSV_VALIDATION_BYTES = 1024
 """Number of bytes read by CSV previewer to validate the file."""
 
+PREVIEWER_CHARDET_BYTES = 1024
+
```
Comment (jirikuncar) [inline invenio_previewer/config.py]: missing docstring

**Comment by slint** [comment]
ping @jirikuncar

**Comment by jirikuncar** [comment]
LGTM - for the future we should include some real files for test cases instead of just mocking the functions to check when exception type changes or the problem is not detected with new version.

**Comment by lnielsen** [comment]
LGTM

**Pull request merged**